File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{   
   let beforeNum = -1
   let start = true
   let data = []
   
   for(const nowNum of arr){  
       
       if(nowNum === beforeNum){
           continue
       }
     
       data.push(nowNum)
       beforeNum = nowNum
    
   }
    return data
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int prevNum = arr[0];
        List<Integer> answer = new ArrayList<>();
        answer.add(prevNum);
        for(int i=1; i<arr.length; i++){
            if(prevNum == arr[i]){
                continue;
            }
            
            answer.add(arr[i]);
            prevNum = arr[i];
        }
        
        
        int[] array = answer.stream().mapToInt(Integer::intValue).toArray();
        
        return array;
    }
}